lighthouse-core/audits/performance-budget.js

```diff
@@ -121,7 +123,11 @@ class ResourceBudget extends Audit {
   static async audit(artifacts, context) {
     const devtoolsLog = artifacts.devtoolsLogs[Audit.DEFAULT_PASS];
     const summary = await ResourceSummary.request({devtoolsLog, URL: artifacts.URL}, context);
-    const budget = context.settings.budgets ? context.settings.budgets[0] : undefined;
+    const mainResource = await MainResource.request({URL: artifacts.URL, devtoolsLog}, context);
+    // Applies the LAST matching budget
+    const budget = context.settings.budgets ? context.settings.budgets.reverse().find((b) => {
```

Comment:
maybe we should copy `budgets` to avoid mutating? maybe a test for this too if we do in fact care about it